core/src/executor/insert.rs

```diff
@@ -178,6 +200,53 @@ async fn fetch_vec_rows<T: GStore>(
     }
 }
 
+async fn validate_foreign_key<T: GStore>(
+    foreign_keys: Vec<ForeignKey>,
+    rows: &[Vec<Value>],
+    storage: &T,
```

Comment:
it would be more consistent to set storage as the first parameter.

Reply:
then, from environmental params to target params
storage, column_defs, foreign_keys and rows